src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -511,14 +521,153 @@ private static bool HasDuplicateInterfaces(NamedTypeSymbol type, ConsList<Symbol
                     set.Free();
                     return false;
             }
-            
+
             // very rare case. 
             // some implemented interfaces are related
             // will have to instantiate interfaces and check
             hasRelatedInterfaces:
             return type.InterfacesNoUseSiteDiagnostics(basesBeingResolved).HasDuplicates(TypeSymbol.EqualsIgnoringDynamicAndTupleNamesComparer);
         }
 
+        /// <summary>
+        /// Deep checking tuple names means checking the constraints on every constituent type and all their base types.
+        /// deepCheckTupleNames should be set to false when the caller is already visiting constituent types.
```

Comment:
>deepCheckTupleNames should be set to false when the caller is already visiting constituent types. [](start = 12, length = 97)

It looks like the comment is somewhat obsolete, doesn't reflect the current implementation.